[CI] Add PR validation on macOS

radical opened the pull request:
This adds PR validation for macOS with github actions. This is at par with Windows tests, with Docker based tests, and tests requiring trusted ssl certificate being disabled.

- **[tests] Run tests workflow on macos**
- **Adjust RequiresDocker and RequiresSSLCertificate to skip macos**
- **Disable tests failing on macos**
  - `FormatDateTime_WithMilliseconds_NewZealandCulture`
  - `AppBar_Change_Theme_ReloadPage`
  - `ProjectWithNoHTTPSRequiresExplicitOverrideWithEnvironmentVariable` 
- **Don't override DOTNET_ROOT when getting list of tests**
- **Skip playground tests on windows and macos**

radical committed:
[tests] Run tests workflow on macos

radical committed:
Adjust RequiresDocker and RequiresSSLCertificate to skip macos

radical committed:
Disable tests failing on macos

radical committed:
Don't override DOTNET_ROOT when getting list of tests

radical committed:
Skip playground tests on windows and macos

.. and add RunOnGithubActionsMacOS

danmoseley reviewed (approved):
LGTM feel free to merge

RussKie commented:
We need to update runsheet runners:
* https://github.com/dotnet/aspire/blob/main/eng/TestRunsheetBuilder/TestRunsheetBuilder.targets
* https://github.com/dotnet/aspire/blob/main/eng/QuarantinedTestRunsheetBuilder/QuarantinedTestRunsheetBuilder.targets

RussKie committed:
Update runsheet runners

RussKie committed:
Merge branch 'main' into macos-ci

RussKie committed:
fixup! Update runsheet runners

RussKie commented on tests/Directory.Build.targets:
```diff
@@ -61,8 +61,9 @@
     <ItemGroup>
-      <TestProject Condition="'$(OS)' == 'Windows_NT'" Include="$(MSBuildProjectFullPath)" RunTestsOnGithubActions="$(RunOnGithubActionsWindows)" />
-      <TestProject Condition="'$(OS)' != 'Windows_NT'" Include="$(MSBuildProjectFullPath)" RunTestsOnGithubActions="$(RunOnGithubActionsLinux)" />
+      <TestProject Condition="$([System.OperatingSystem]::IsWindows())" Include="$(MSBuildProjectFullPath)" RunTestsOnGithubActions="$(RunOnGithubActionsWindows)" />
+      <TestProject Condition="$([System.OperatingSystem]::IsLinux())" Include="$(MSBuildProjectFullPath)" RunTestsOnGithubActions="$(RunOnGithubActionsLinux)" />
+      <TestProject Condition="$([System.OperatingSystem]::IsMacOS())" Include="$(MSBuildProjectFullPath)" RunTestsOnGithubActions="$(RunOnGithubActionsMacOS)" />
```
Please use `BuildOS` property instead:
https://github.com/dotnet/aspire/blob/35fbfd37d71176c73ba61d6a06a9e54ac3319507/Directory.Build.props#L45-L47

RussKie commented:
* https://github.com/dotnet/aspire/actions/runs/14918382872

RussKie commented:
Outerloop is unhappy, looking

RussKie committed:
fixup! Update runsheet runners

RussKie commented:
Do we need to run E2E on a mac? Those seem to time-out...

RussKie commented on tests/Directory.Build.targets:
```diff
@@ -36,7 +36,7 @@
            Text="%24(ExtractTestClassNamesPrefix) should be set, for example - Aspire.Templates.Tests" />
 
-    <Exec Command="&quot;$(RunCommand)&quot; --filter-not-trait category=failing --list-tests" ConsoleToMSBuild="true" EnvironmentVariables="DOTNET_ROOT=$(DotNetRoot);DOTNET_ROOT_X86=$(DotNetRoot)x86">
+    <Exec Command="&quot;$(RunCommand)&quot; --filter-not-trait category=failing --list-tests" ConsoleToMSBuild="true">
```
May you please provide the reason behind this change?

radical replied:
If you don't agree with this btw, then we can discuss and fix in a follow up.

RussKie replied:
@Youssef1313

radical committed:
Merge remote-tracking branch 'origin/main' into macos-ci

radical committed:
Address review feedback from @ russkie

radical committed:
Merge remote-tracking branch 'origin/main' into macos-ci

radical committed:
Address review feedback from @ russkie - skip E2E tests on macos as they

.. are timing out.

Issue: https://github.com/dotnet/aspire/issues/9198

radical committed:
Add missing CI: false for outerloop test runs

Fixes
```
/home/runner/work/aspire/aspire/eng/Versions.targets(49,67): error MSB4057: The target "GitInfo" does not exist in the project. [/home/runner/work/aspire/aspire/src/Microsoft.Extensions.ServiceDiscovery.Abstractions/Microsoft.Extensions.ServiceDiscovery.Abstractions.csproj]
/home/runner/work/aspire/aspire/eng/Versions.targets(49,67): error MSB4057: The target "GitInfo" does not exist in the project. [/home/runner/work/aspire/aspire/src/Microsoft.Extensions.ServiceDiscovery.Abstractions/Microsoft.Extensions.ServiceDiscovery.Abstractions.csproj]
/home/runner/work/aspire/aspire/eng/Versions.targets(49,67): error MSB4057: The target "GitInfo" does not exist in the project. [/home/runner/work/aspire/aspire/src/Tools/ConfigurationSchemaGenerator/ConfigurationSchemaGenerator.csproj]
/home/runner/work/aspire/aspire/eng/Versions.targets(49,67): error MSB4057: The target "GitInfo" does not exist in the project. [/home/runner/work/aspire/aspire/src/Microsoft.Extensions.ServiceDiscovery.Abstractions/Microsoft.Extensions.ServiceDiscovery.Abstractions.csproj]
/home/runner/work/aspire/aspire/eng/Versions.targets(49,67): error MSB4057: The target "GitInfo" does not exist in the project. [/home/runner/work/aspire/aspire/src/Tools/ConfigurationSchemaGenerator/ConfigurationSchemaGenerator.csproj]
```

radical committed:
Add macos to the quarantined tests report generator

radical committed:
Merge remote-tracking branch 'origin/main' into macos-ci
